event-based-simulators/extras/ExportProfileData.py

```diff
@@ -0,0 +1,213 @@
+    sys.exit()
+######################################################
+
+
+def ExportAllProfileDataFromChildDevices(createFrom, createTo):
+    deviceInTenantCount = 0
+    deviceManagedObject = c8y.device_inventory.select(type=C8Y_OEE_SIMULATOR_DEVICES_GROUP)
+    for device in deviceManagedObject:
+        deviceInTenantCount += 1
+        LogDebug(f"Found device '{device.name}', id: #{device.id}, owned by {device.owner}, number of children: {len(device.child_devices)}, type: {device.type}")
+        LogDebug(f"List of {device.name}'s child devices: ")
+        for childDevice in device.child_devices:
+            ExportSpecificProfileDataWithDeviceId(createFrom=createFrom,createTo=createTo, deviceId=childDevice.id)
+
+    if deviceInTenantCount == 0:
+        LogInfo(f"No device in tenant {c8y.tenant_id} found")
+
+
+def ExportSpecificProfileDataWithDeviceId(createFrom, createTo, deviceId):
+    deviceName = FindDeviceNameById(deviceId, c8y.base_url)
+    deviceWithIdCount = 0
+    deviceExternalId, deviceExternalIdType = CheckDeviceExternalIdById(deviceId, c8y.base_url)
+    if not deviceExternalId:
+        return
+    if IsExternalIdTypeEventBasedSimulatorProfile(deviceExternalIdType):
+        filePath = CreateFilePath(Id=deviceExternalId)
+    else:
+        return
+    LogDebug(f"Search for {DATA_TYPE} data from device {deviceName}, id #{deviceId}")
+    for device in c8y.device_inventory.select(name=deviceName):
+        deviceWithIdCount += 1
+        fileLogger.debug(f"Child device {device.name}, id #{device.id}")
+        if DATA_TYPE == "alarms":
+            ExportAlarms(device, createFrom, createTo, filePath)
+            AppendDataToJsonFile([], filePath, 'measurements')
+            fileLogger.debug(f"{DATA_TYPE.capitalize()} data is added to data file at {filePath}")
+        elif DATA_TYPE == "measurements":
+            # listing measurements of child device
+            ExportMeasurements(device, createFrom, createTo, filePath)
+            AppendDataToJsonFile([], filePath, 'alarms')
+            fileLogger.debug(f"{DATA_TYPE.capitalize()} data is added to data file at {filePath}")
+        else:
+            ExportAlarms(device, createFrom, createTo, filePath)
+            ExportMeasurements(device, createFrom, createTo, filePath)
+            fileLogger.debug(f"Alarms and Measurements data is added to data file at {filePath}")
+
+    if deviceWithIdCount == 0:
+        fileLogger.info(f"No data for device with id #{deviceId} is exported")
+    else:
+        LogInfo(f"Exported successfully data for device with id #{deviceId}, external id: {deviceExternalId}")
+    return
+
+
+def FindDeviceNameById(deviceId, baseUrl):
+    response = requests.get(f'{baseUrl}/inventory/managedObjects/{deviceId}',
+                            headers=C8Y_HEADERS)
+    if not response.ok:
+        LogError(f"Connection to url '{baseUrl}/inventory/managedObjects/{deviceId}' failed. Check your parameters in environment file again")
```

Comment:
I'd suggest to also print the response code/text in case of an error as it helps to determine the problem.